e2e: node: cpumanager: replace the test suite

ffromani opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Replace the old cpumanager tests with the new suites rewritten and tested through the full 1.34 cycle.
The new testsuite is a superset of the old suite, so we gain coverage.

#### Which issue(s) this PR is related to:
https://github.com/kubernetes/kubernetes/issues/129884

#### Special notes for your reviewer:
This PR depends on and requires https://github.com/kubernetes/kubernetes/pull/133462

Previous work
- https://github.com/kubernetes/kubernetes/pull/130274
- https://github.com/kubernetes/kubernetes/pull/131794
- https://github.com/kubernetes/kubernetes/pull/132077
- https://github.com/kubernetes/kubernetes/pull/132215
- https://github.com/kubernetes/kubernetes/pull/132498
- https://github.com/kubernetes/kubernetes/pull/132811
- https://github.com/kubernetes/kubernetes/pull/132818
- https://github.com/kubernetes/kubernetes/pull/133462

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

ffromani committed:
e2e: node: cpumgr: remove old cfs quota tests

superseded by code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
e2e: node: cpumgr: remove old smt alignment tests

superseded by code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
e2e: node: cpumgr: remove old distribute-cpus tests

superseded by code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
e2e: node: cpumgr: remove old reserved cpus tests

superseded by code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
e2e: node: cpumgr: remove old sidecar container tests

superseded by code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
e2e: node: cpumgr: keep only scaffolding

keep only the test stub, as all the code was already
removed by PR in the series because superseded by
code in cpumanager_test.go,
which will be moved in cpu_manager_test.go at the
end of this series.
Split to make the review easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

matthyx commented:
/lgtm

ffromani commented:
/hold

Sorry, I failed to point out clearly this PR depends on https://github.com/kubernetes/kubernetes/pull/133462, fixing now

pacoxu commented:
> /hold
> 
> Sorry, I failed to point out clearly this PR depends on #133462, fixing now

https://github.com/kubernetes/kubernetes/pull/133462 was merged. Should this be rebased for next step?

matthyx commented:
@ffromani ping me when you are ready

ffromani committed:
e2e: node: cpumgr: replace old testsuite

This final change in the series completes the transition
to the new test suite

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
that's it! rebased and this is the final PR in this long journey

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-topology-manager
/test pull-kubernetes-node-kubelet-serial-topology-manager-kubetest2
/test pull-kubernetes-node-kubelet-serial-memory-manager
/test pull-kubernetes-node-kubelet-serial-cpu-manager
/test pull-kubernetes-node-kubelet-serial-cpu-manager-kubetest2
/test pull-crio-cgroupv1-node-e2e-resource-managers
/test pull-crio-cgroupv2-node-e2e-resource-managers

ffromani commented:
reviewers please be aware that this PR did include the content of https://github.com/kubernetes/kubernetes/pull/133462 . Now that https://github.com/kubernetes/kubernetes/pull/133462 merged, I rebased to include only the last changes needed, which is the removal of the old tests replaced by the rewritten tests.

ffromani commented:
/hold cancel
/triage accepted

(triage accepted from the context)

ffromani commented:
/test pull-crio-cgroupv2-node-e2e-resource-managers
/test pull-crio-cgroupv1-node-e2e-resource-managers

ffromani commented:
@matthyx howdy, we're ready for review!

matthyx reviewed (approved):
/lgtm

ffromani committed:
e2e: node: address linter errors

remove now-unused code

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani commented:
/test pull-kubernetes-node-kubelet-serial-topology-manager
/test pull-kubernetes-node-kubelet-serial-topology-manager-kubetest2
/test pull-kubernetes-node-kubelet-serial-memory-manager
/test pull-kubernetes-node-kubelet-serial-cpu-manager
/test pull-kubernetes-node-kubelet-serial-cpu-manager-kubetest2
/test pull-crio-cgroupv1-node-e2e-resource-managers
/test pull-crio-cgroupv2-node-e2e-resource-managers

ffromani commented:
/retest-required
